### PR TITLE
Improved JDK annotations

### PR DESCRIPTION
**Pull request by Maxi17**
Some improvements to offsets in java.io and java.lang. Some new files are also included.

### CODE REVIEW

File: checker/jdk/index/src/java/nio/channels/ByteChannel.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/*
```
Comment (kelloggm) [inline checker/jdk/index/src/java/nio/channels/ByteChannel.java]: What is the purpose of including files that have no annotations in them?

**Comment by kelloggm** [comment]
However, there is a merge conflict, which needs to be resolved before this PR can be merged.

File: checker/jdk/index/src/java/nio/channels/ByteChannel.java (hunk 1 above)
Comment (Maxi17) [reply checker/jdk/index/src/java/nio/channels/ByteChannel.java]: When I added one class, I thought it is a good idea for completeness to add its superclasses/subclasses, or other related classes.

Comment (kelloggm) [reply checker/jdk/index/src/java/nio/channels/ByteChannel.java]: Okay, that is reasonable.

File: checker/build.gradle [hunk 2]
```diff
@@ -11,7 +11,7 @@ sourceSets {
     main {
         resources {
             // Stub files, message.properties, etc.
-            srcDirs += ['src/main/java']
+            srcDirs = ['src/main/java']
```
Comment (kelloggm) [inline checker/build.gradle]: This is an odd change that doesn't seem to have anything to do with the rest of this PR. What's the reasoning behind it? I don't think I can merge this in good conscience without knowing why this change is needed.

**Pull request merged**